src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -226,20 +226,20 @@ winrt::Microsoft::Terminal::Settings::Model::Profile CascadiaSettings::CreateNew
         return nullptr;
     }
 
-    winrt::hstring newName{};
-    for (uint32_t candidateIndex = 0; candidateIndex < _allProfiles.Size() + 1; candidateIndex++)
+    std::wstring newName;
+    for (uint32_t candidateIndex = 0, count = _allProfiles.Size() + 1; candidateIndex < count; candidateIndex++)

```

Comment:
Can/should you `const uint32_t` the `count` initialization?

Also... multiple initializations in for loops always confuse me. But I'm not willing to block over it.

Reply:
Unfortunately you cannot have multiple different types in for loops in C++.
You can only write something like `for (auto [a, b, ...] = std::tuple{ A, B, ... }; ...`

Reply:
Sad. Whatever. I'll survive.